packages/rum-core/src/browser/performanceCollection.ts

```diff
@@ -250,19 +250,24 @@ function retrieveFirstInputTiming(callback: (timing: RumFirstInputTiming) => voi
   }
 }
 
-interface IndexedPerformanceTiming extends PerformanceTiming {
-  [key: string]: any
+type IndexedPerformanceTiming = {
+  [key in keyof PerformanceTiming]: any
+}
+
+export type RelativePerformanceTiming = {
+  -readonly [key in keyof Omit<PerformanceTiming, 'toJSON'>]: RelativeTime
 }
 
 function computeRelativePerformanceTiming() {
-  const result: Partial<IndexedPerformanceTiming> = {}
+  const result: Partial<RelativePerformanceTiming> = {}
   const timing = performance.timing as IndexedPerformanceTiming
```

Comment:
Do we really need this cast? to my understanding it only makes the type of `timing[numberKey]` bellow `any` instead of `number`.

Reply:
Indeed, by being more explicit about used keys, we don't need to have this IndexedType anymore 👍